### ISSUE TITLE
Remove the incorrect warning "The package is missing a Setup.hs or Setup.lhs script."

### ISSUE THREAD

**Issue by nikita-volkov** [haskell/cabal#2629]
```
$ cabal sdist
Distribution quality errors:
The package is missing a Setup.hs or Setup.lhs script.
Note: the public hackage server would reject this package.
```

What's stated there is simply untrue: the public hackage server does not reject such packages. It's been there forever, but this time it got annoying.

**Comment by 23Skidoo** [comment]
It's not incorrect for build types other than `Simple`.

**Comment by nikita-volkov** [comment]
Well, okay. But it still is incorrect in case of the `Simple` build type. Can't it be fixed with a precondition?

**Comment by 23Skidoo** [comment]
Fixed by #2657.
